src/Model/Interactive.php

```diff
@@ -58,27 +59,55 @@ class Interactive extends DataObject {
         'TrackViews'        => 'Varchar(16)',
 
         'CompletionElement'   => 'Varchar(64)',         // what element needs clicking to be considered a 'complete' event
+    ];
 
-	);
-
-	private static $has_one = array(
+	private static $has_one = [
 		'InternalPage'		=> 'Page',
 		'Campaign'			=> InteractiveCampaign::class,
 		'Image'				=> Image::class,
-    );
+    ];
 
     private static $owns = [
         'Image',
     ];
 
-    private static $extensions = array(
+    private static $extensions = [
         InteractiveLocationExtension::class,
         Versioned::class
-    );
+    ];
 
     private static $tracker_type = 'Local';
 
-    private static $summary_fields = array('Title', 'Clicks', 'Impressions', 'Completes');
+    private static $summary_fields = ['Title', 'Clicks', 'Impressions', 'Completes'];
+
+    public function populateDefaults()
+    {
+        // frequency
+        if ($freq = Config::inst()->get(self::class, 'Frequency')) {
+            $this->Frequency = $freq;
+        } else {
+            $this->Frequency = 1;
+        }
+        // delay
+        if ($delay = Config::inst()->get(self::class, 'Delay')) {
```

Comment:
Can you use `self::config()->Delay` for this instead of an explicit call to `Config::`? (obviously, if we can swap to $defaults, this point is moot

Reply:
Done